hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlepubsub/CompressionCodecFactory.java

```diff
@@ -84,7 +84,7 @@ static class ZstandardCodecFactory extends CompressionCodecFactory {
 
         @Override
         Codec createInstance() {
-            return new ZstandardCodec(compressionLevel, true);
+            return new ZstandardCodec(compressionLevel, true, ZstandardCodec.DEFAULT_USE_BUFFERPOOL);
```

Comment:
were we using buffer pool previously or not?

Reply:
Probably not, as the support for the buffer pool was added in 1.10.X (we previously used 1.9.X) version. Here is the PR with added support for buffer pool in apache avro library: https://github.com/apache/avro/commit/cff3bdad6e32faf74cae7321e0bcef83846d5351. 
The previous apache version was using 
![Screenshot 2024-04-29 at 14 11 11](https://github.com/allegro/hermes/assets/11329819/aec1c9b1-2ff9-4c86-9f68-3b15bb6d9524)
`new ZstdOutputStream(compressed, bounded)` constructor which was like:
<img width="773" alt="Screenshot 2024-04-29 at 14 13 30" src="https://github.com/allegro/hermes/assets/11329819/535da086-ae0b-4ba7-ac30-923dbd15be00">

Reply:
I picked the default pool option as a sensible default (meaning no pooling)